### DISCUSSION TITLE
axios can't receive error response status code

### DISCUSSION THREAD

**Discussion by 2277509846** [fastapi/fastapi#11212] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

Here is a middleware I defined:
class AccessControlMiddleware(BaseHTTPMiddleware):  
    async def dispatch(self, request: Request, call_next: Callable[..., Awaitable[Response]]) -> Response:  
        if request.url.path == '/api/forbidden':
            return Response(status_code=status.HTTP_403_FORBIDDEN, content="Forbidden")
        response = await call_next(request)
        return response

When I access an interface that does not exist, axios can know the 404 status code from the response, but when I access /api/forbidden, I cannot know the status code.

axios.get('http://127.0.0.1:8000/api/notfound').then(res => {
  console.log(res);
}).catch(err => {
  console.log(err.response); // axios error contains 404
});

axios.get('http://127.0.0.1:8000/api/forbidden').then(res => {
  console.log(res);
}).catch(err => {
  console.log(err.response); // undefined
})

**Comment by jgould22** [comment, marked as answer]
When you use Github to post code examples try and use the[ triple back ticks ](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code)around your code so it is easier for others to read. It increases the chance of getting a response 

Anyway, when you make a request to FastAPI and you do not end your request with a slash, like `/api/forbidden` FastAPI will respond with an HTTP 307 redirect and tell axios it should actually go to  `/api/forbidden/` (note the slash at the end). Axios will follow this automatically and go to  `/api/forbidden/` . This means you are checking for `/api/forbidden` but the request is actually for `/api/forbidden/`.

Either way step one is to use some logging and see what the value is in `request.url.path`
